### PR TITLE
swap: replaced Owner for Issuer

### PR DESCRIPTION
**Pull request by santicomp2014**
Resolves Issue #1553 
Changes Owner for Issuer, So the terminology is the same as the Smart Contract.

### CODE REVIEW

File: swap/cheque.go
Comment (mortelli) [inline swap/cheque.go]: since a contract is not issued (unlike a cheque) i don't think the terminology change should be applied in this case

File: swap/swap_test.go
Comment (mortelli) [inline swap/swap_test.go:860]: same as above: not convinced `swap` or `contract` can use the terminology of `issued` like cheques can

**Review by mortelli — CHANGES_REQUESTED** [review]
i am leaving a general comment here in order not to spam, but my overall impression is that we shouldn't use the term `issuer` when talking about either the `Swap` structure or making direct references to contracts (instead of cheques). 

i am aware that issue #1553 does not state this, but i am objecting to it; to me, `owner` still feels better suited to these cases.

tagging @Eknir as well as @ralph-pichler since they are the ones most familiar with the terminology used in the SCs. do they use the term `owner` at all?

i will approve the PR if we are not in agreement on this and a case can be made for the opposing view :+1:

File: swap/cheque.go
Comment (ralph-pichler) [reply swap/cheque.go]: the contract does not use the `owner` terminology anymore. But you are correct that the contract is not issued (the "issuer" is the party allowed to issue cheques but not the "issuer" of the contract). `allowed issuer` or `issuing party` might be some option, but I think outside the context of a cheque we can also keep it as `owner` on the go side.

**Review by ralph-pichler — COMMENTED** [review]
"Simplest" Swap has landed on master. Please rebase.

File: swap/swap.go
Comment (Eknir) [inline swap/swap.go]: You could make it: `getIssuerAtContract`, or decide that we can cope with some ambiguity here.

File: swap/swap_test.go
Comment (Eknir) [reply swap/swap_test.go:860]: The confusion here has in my opinion not so much to do with renaming `owner` to `issuer`. 

It underlines, IMO, that we have created somewhat of confusin situation due to naming:

Here we see again that an issuer can be also a beneficiary and vice versa. What makes it extra confusing here, is that in the Swarm source code (`protocol.go:124` and `peer.go:34`), we speak about the `peer.beneficiary` as the counterparty and we set our own address and contract in the `swap` struct. Whereas here, we are just doing it the other way around, saving the `issuer` in the peer struct and the beneficiary in the `swap` struct.

We have 4 parties to name properly:

- Counterparty Externally Owned account,
Currently `peer.beneficiary`
- Counterparty chequebook address,
Currently `peer.contractAddress`
- Our own Externally Owned account,
Currently `swap.issuer.address`
- Our own chequebook address
Currently `swap.issuer.Contract`

What do you think: is it worth spending time on attempting to find perfect names for those, or shall we just leave as it is, and accept that there can be ambiguity and that we risk making mistakes as in the function `newTestSwapAndPeer`?

**Review by Eknir — APPROVED** [review]
Good first PR and sorry for the long comments!

I generally have the feeling, as expressed in the comments, that our naming is not always spot on and that this might create confusion. I remember though the wise words from @mortelli (ask him about this) and perhaps we should just move on and don't attempt to make the naming better.

Nevertheless, I think it is an improvement to rename `issuer` to `owner`, as `owner` is not used anymore on the smart-contract side. 

I approve the PR, as I think it is good, as is. We could perhaps make it better, but then I would like to assess the whole issue with regards to naming and not only talk about renaming `owner` to `issuer`.

File: swap/cheque.go
Comment (santicomp2014) [reply swap/cheque.go]: I will commit the changed comment, `the beneficiary is the issuer of cheques of the counterparty swap contract`.

I replaced all instances just to be on the safe side, but I agree it depends on the context.
I agree that it's a bit confusing.

File: swap/swap.go
Comment (santicomp2014) [reply swap/swap.go]: It makes more sense this way.

File: swap/swap_test.go
Comment (santicomp2014) [reply swap/swap_test.go:860]: I understand the cause of confusion, should I leave alone for now?.

Or should I refactor the comments?, I'm a bit confused.

File: swap/protocol.go
Comment (holisticode) [inline swap/protocol.go:108]: This is exactly what I meant this morning in the call. 

This is the handshake. At this point, two nodes are just exchanging contract addresses. We don't even know what balances there are (they might be 0 on both sides at this point!). Thus, I see it as incorrect to change this to `issuer`. No cheques may have been issued at this point, thus no issuer exists. Furthermore, at this point both nodes could be the issuer of cheques! I think it does not make sense to introduce this change here.

File: swap/swap_test.go
Comment (Eknir) [reply swap/swap_test.go:860]: I would leave this one for now. I will bring it up as a point on the agenda for our next sync meeting!

File: swap/cheque.go
Comment (mortelli) [inline swap/cheque.go:120]: so if i understand this correctly, the mentioned `counterparty` here would be the peer `p`, right?

i find it confusing to call this `beneficiary`, as it could be interpreted as _the address this peer writes cheques to_ or _the address this peer wants its cheques written to_ just as well.

also: remove `.` from this comment

File: swap/protocol.go
Comment (mortelli) [reply swap/protocol.go:108]: i agree. 

if we don't want to use `owner` for reasons of consistency with the smart contracts, we can find another word.

in any case, the golang context is different from the SC context, so it wouldn't bother me if we were to use additional terms on either side that the other side doesn't.

i'd just insist on using the same terms when referring to the same entities, specially in variables or fields.

File: swap/protocol_test.go
Comment (mortelli) [inline swap/protocol_test.go:50]: same here, as commented above (i will refrain from repeating it from here onwards)

**Review by mortelli — COMMENTED** [review]
i have to say that at this point, i am not sure of how to move forward here. 

i cannot approve this PR, nor request changes, since it seems many of us have different opinions on what each thing should be named.

i am officially requesting commentary/thoughts on how to proceed in this case: from @holisticode due to being the track lead, and/or @vojtechsimetka due to being the issue creator.

File: swap/cheque.go
Comment (Eknir) [reply swap/cheque.go:120]: Agree about `beneficiary` being confusing. I would change it to `p.counterparty`

**Comment by santicomp2014** [comment]
Owner to Issuer is not agreed on by consensus.
So for the time being this will be on hold until a better nomenclature comes along.

**Comment by Eknir** [comment]
Can we close this PR and address the proposed changes here in the upcoming rename PR? 
@santicomp2014

**Comment by santicomp2014** [comment]
This PR was closed due to the outdated changes between this code and the latest additions of ERC20 and Swap Factory.

It makes more sense to branch of master, and then include the comments that were provided in this PR.

**Pull request closed without merging**
